qa/rpc-tests/wallet.py

```diff
@@ -463,6 +463,62 @@ def run_test (self):
 
         assert_equal("not an integer" in errorString, True);
 
+        myzaddr     = self.nodes[0].z_getnewaddress()
+        recipients  = [ {"address": myzaddr, "amount": Decimal('0.0') } ]
+        errorString = ''
+
+        # Make sure that amount=0 transactions can use the default fee
+        # without triggering "absurd fee" errors
+        try:
+            myopid = self.nodes[0].z_sendmany(myzaddr, recipients)
+            assert(myopid)
+        except JSONRPCException,e:
+            errorString = e.error['message']
+            print errorString
+            assert(False)
+
+        # This fee is larger than the default fee and since amount=0
+        # it should trigger error
+        fee         = Decimal('0.1')
+        recipients  = [ {"address": myzaddr, "amount": Decimal('0.0') } ]
+        minconf     = 1
+        errorString = ''
+
+        try:
+            myopid = self.nodes[0].z_sendmany(myzaddr, recipients, minconf, fee)
+        except JSONRPCException,e:
+            errorString = e.error['message']
+            assert('Small transaction amount' in errorString)
```

Comment:
This assert should be moved to the outer block, since we expect an error and we want to assert the error string has been set. (I will push a commit to do this).